src/codegen/expression.rs

```diff
@@ -3757,3 +3828,186 @@ fn add_prefix_and_delimiter_to_print(mut expr: Expression) -> Expression {
         }
     }
 }
+
+/// Check the return code of `transfer`.
+///
+/// If `bubble_up` is set to true, this will revert the contract execution on failure.
+/// Otherwise, the comparison if the return code equals `0` is returned.
+fn polkadot_check_transfer_ret(
+    loc: &Loc,
+    success: usize,
+    cfg: &mut ControlFlowGraph,
+    ns: &Namespace,
+    opt: &Options,
+    vartab: &mut Vartable,
+    bubble_up: bool,
+) -> Option<Expression> {
+    let ret_code = Expression::Variable {
+        loc: *loc,
+        ty: Type::Uint(32),
+        var_no: success,
+    };
+    let ret_ok = Expression::NumberLiteral {
+        loc: *loc,
+        ty: Type::Uint(32),
+        value: 0.into(),
+    };
+    let cond = Expression::Equal {
+        loc: *loc,
+        left: ret_code.into(),
+        right: ret_ok.into(),
+    };
+
+    if !bubble_up {
+        return Some(cond);
+    }
+
+    let success_block = cfg.new_basic_block("transfer_success".into());
+    let fail_block = cfg.new_basic_block("transfer_fail".into());
+    cfg.add(
+        vartab,
+        Instr::BranchCond {
+            cond,
+            true_block: success_block,
+            false_block: fail_block,
+        },
+    );
+
+    cfg.set_basic_block(fail_block);
+    let msg = "value transfer failure";
+    log_runtime_error(opt.log_runtime_errors, msg, *loc, cfg, vartab, ns);
+    cfg.add(vartab, Instr::AssertFailure { encoded_args: None });
+
+    cfg.set_basic_block(success_block);
+
+    None
+}
+
+pub(super) mod polkadot {
```

Comment:
Why do we need a new module here? I think it should be in a new file if it needs to be in a module.

Reply:
A new file is a good idea.